packages/jaeger-ui/src/components/SearchTracePage/index.js

```diff
@@ -63,12 +67,23 @@ export class SearchTracePageImpl extends Component {
       fetchMultipleTraces(needForDiffs);
     }
     fetchServices();
+    if (servicePollingEnabled === true) {
+      this.pollServicesTimer = setInterval(() => {
+        fetchServices();
```

Comment:
Unfortunately this change will cause the UI to re-render twice per interval. [`fetchStarted` will set `state.services.loading` to `true`](https://github.com/jaegertracing/jaeger-ui/blob/master/packages/jaeger-ui/src/reducers/services.js#L29) when the long polling request is raised, then [`fetchDone` will set `state.services.loading` to `false`](https://github.com/jaegertracing/jaeger-ui/blob/master/packages/jaeger-ui/src/reducers/services.js#L35) even if nothing changed between the requests.

`fetchStarted` should probably only set `loading` to `true` if `state.services` is `null` or `[]` else return the existing state:
```
if (state.services && state.services.length) return state;
return { ...state, loading: true };
```

and `fetchDone` should only change the state if it was previously loading or if the services have changed (fortunately `services` are sorted anyway so this check won't increase time complexity):
```
const services = payload.data || [];
services.sort(localeStringComparator);
if (state.loading || !state.services || services.length !== state.services.length || services.some((service, i) => service !== state.services[i]) {
  return { ...state, services, error: null, loading: false };
}
return state;
```